tests/MongoDB.Driver.Tests/Specifications/command-monitoring/TestRunner.cs

```diff
@@ -105,104 +90,169 @@ public void RunTestDefinition(IEnumerable<BsonDocument> data, string databaseNam
         {
             definition = (BsonDocument)DeepCopy(definition); // protect against side effects when the same definition is run twice (async=false/true)
 
-            BsonValue bsonValue;
-            if (definition.TryGetValue("ignore_if_server_version_greater_than", out bsonValue))
+            using (var client = CreateDisposableClient(definition, __capturedEvents))
```

Comment:
this spec tests also require explicit setting `retryWrites` to `false`.
The only change here is replacing the global `__client` variable on creating `disposable` client for each test and parsing `clientOptions` section.